shared/js/event_dispatcher.js

```diff
@@ -199,7 +215,7 @@
             'Object to mix into already has "' + method + '" property defined!'
           );
         }
-        target[method] = eventDispatcher[method].bind(this);
+        this[method] = target[method] = eventDispatcher[method].bind(this);
```

Comment:
I did this in the past: https://github.com/julienw/studio/blob/autotheme/js/libs/event_dispatcher.js#L103-L110
(never filed a bug to land it, sorry :p)

basically calling eventDispatcher.on.call(this...).